tutorials/darcy_thermo_mech/doc/content/workshop/numerical/fvm_overview.md

```diff
@@ -0,0 +1,245 @@
+
+!equation
+\int_\Omega cu ~dV = \sum_i^{N_c}\int_{\Omega_i} c u ~dV, \quad \text{and} \quad \int_\Omega f ~dV = \sum_i^{N_c}\int_{\Omega_i} f ~dV
+
+Use the finite volume approximation (on cell C):
+
+!equation
+\int_{\Omega_C} c u ~dV\approx c_Cu_CV_C, \quad \text{and} \quad \int_{\Omega_C} f ~dV \approx f_C V_C
+
+!---
+
+## Approximating the Reaction and Source Terms
+
+For a finite volume variable, there is only one quadrature point and it is the cell/face centroid.
+
+!listing framework/src/fvkernels/FVReaction.C re=ADReal\sFVReaction::computeQpResidual.*?^}
+
+!---
+
+## Approximating the Diffusion Term
+
+!row!
+!col! width=40%
+
+!style fontsize=80%
+Othogonal scenario: ($S_f$ and $\delta_{NP}$ are parallel)
+
+!media darcy_thermo_mech/fvm_orthogonal.png style=width:100%;background:white;
+
+!style fontsize=80%
+Non-othogonal scenario:
+
+!media darcy_thermo_mech/fvm_nonorthogonal.png style=width:100%;background:white;
+
+!col-end!
+
+!col! width=60%
+
+!style fontsize=80%
+Split into cell-wise integrals and use the divergence theorem:
+
+!equation
+\int_\Omega\nabla\cdot k\nabla u dV = \sum_i^{N_c}\int_{\Omega_i}\nabla\cdot k\nabla u dV = \sum_i^{N_c}\int_{\partial\Omega_i} k\nabla u \cdot \hat{n} dS
+
+!style fontsize=80%
+On an internal cell (let's say on cell C):
+
+!equation
+\int_{\partial\Omega_C} k\nabla u \cdot \hat{n} dS = \sum_f^{N_{f,C}}\int_{\partial\Omega_{C,f}}k\nabla u \cdot \hat{n} dS \approx \sum_f^{N_{f,C}}k_f\left(\nabla u\right)_f \cdot \hat{n}_f |S_f|
+
+!style fontsize=80%
+- $\left(\nabla u\right)_f \cdot \hat{n}$ for orthogonal scenario: $\frac{u_N-u_C}{\delta_{CN}}$ (cheap and accurate)
+
+!style fontsize=80%
+- $\left(\nabla u\right)_f \cdot \hat{n}$ for nonorthogonal scenario: $\frac{u_N-u_C}{\delta_{CN}}|\Delta_f|+\hat{k}_f\cdot \left(\nabla u\right)_f$  
+
+!style fontsize=80%
+The surface normal ($\hat{n}$) is expanded into a $\delta_{CN}$-parallel ($\hat{\Delta}_f$) and a remaining vector ($\hat{k}_f$): $\hat{n}_f=\hat{\Delta}_f + \hat{k}_f$
+
+!col-end!
+
+!row-end!
+
+!---
+
+## Approximating the Diffusion Term
+
+!listing framework/src/fvkernels/FVDiffusion.C re=ADReal\sFVDiffusion::computeQpResidual.*?^}
+
+!---
+
+## Interpolation and Gradient Estimation
+
+!row!
+!col! width=40%
+
+!style fontsize=80%
+Basic interpolation scheme:
+
+!media darcy_thermo_mech/fvm_interpolation.png style=width:100%;background:white;
+
+!equation
+u_f = \left(\frac{\delta_{fN}}{\delta_{CN}}\right) u_C + \left(1- \frac{\delta_{fN}}{\delta_{CN}}\right) u_N
+
+!col-end!
+
+!col! width=60%
+
+!style fontsize=80%
+Cell gradients can be estimated using the Green-Gauss theorem:
+
+!equation
+\int_{\Omega_i}\nabla u ~dV \approx (\nabla u)_i V_i
+
+!style fontsize=80%
+At the same time:
+
+!equation
+\int_{\Omega_i}\nabla u ~dV = \int_{\partial\Omega_i}u \hat{n} dS \approx \sum_f u_f \hat{n}_f |S_f|
+
+!style fontsize=80%
+From the two equations:  
+
+!equation
+(\nabla u)_i = \frac{1}{V_i} \sum_f u_f \hat{n}_f |S_f|
```

Comment:
Shall we mention that the \left(\nabla u\right)_f from two slides before is computed interpolating the cell gradients?

Reply:
Restructured and added a note!